Update GWP to include `AR6GWP100` & `globalwarmingpotentials` package

danielhuppmann opened the pull request:
This PR removes the local copy of the GWP-conversion factors and adapts the update script to use the [globalwarmingpotentials](https://github.com/openclimatedata/globalwarmingpotentials) package as a community resource for compiling such data.

The PR should be redirected to https://github.com/IAMconsortium/units after:
 - [x] Merging of https://github.com/IAMconsortium/units/pull/28
 - [x] AR6-GWP factors are added to the globalwarmingpotentials package
 - [ ] [maybe] AR6-GWP20 factors are added to the globalwarmingpotentials package https://github.com/openclimatedata/globalwarmingpotentials/pull/12
 - [x] The update-script is run with a stable release of globalwarmingpotentials

Switching to the globalwarmingpotentials also revealed that some mistake must have happened when initially migrating the GWP csv file into this repo, similar to what's discussed in https://github.com/openclimatedata/globalwarmingpotentials/issues/2. This PR corrects the mistake and - by depending on a dedicated package - reduces the likelihood that it occurs again...

khaeru commented on iam_units/data/emissions/SARGWP100.txt:
```diff
@@ -44,18 +43,6 @@
     a_HFC32 = 650.0
     a_HFC41 = 150.0
     a_HFC4310mee = 1300.0
-    a_HFE125 = 14900.0
```
Just to confirm & make explicit the mistake mentioned in the PR description & linked issue:

The source currently cited in our README (https://www.ghgprotocol.org/sites/default/files/ghgp/Global-Warming-Potential-Values%20%28Feb%2016%202016%29_1.pdf) includes this species (and those below) for AR4 and AR5, but not for SAR (the cell is blank).

The file metric_conversions.csv (initially reused from openscm-units, and now deleted in this PR), included the line:
```
Species,Symbol,SARGWP100,AR4GWP100,AR5GWP100
…
,HFE125,14900,12400,
```

So that was actually an erroneous transcription of the table, shifted left by one cell. The correct line should have been:
```
,HFE125,,14900,12400
 ```

In short, the removed lines in this generated file shouldn't have been there in the first place.

khaeru reviewed (changes requested):
Please update the documentation:
- README.rst, where the source(s) used for GWPs are mentioned and linked.
- DEVELOPING.rst, to make explicit the package(s) that must be installed in order to regenerate the Pint context files.
  (This could be added as an "extras" requirement named "update" in setup.cfg, but we could also postpone that for another PR.)

danielhuppmann commented:
Thanks for the comments @khaeru - made me realize that I had overlooked one needed to also extend the list of metrics in `emissions.txt` - so I also refactored that to be written automatically as a new file as part of `update`.

I updated the readme, the developer guidelines and added `globalwarmingpotentials` as an extra-dependency.

Also, I added tests for the new metrics, which failed because Quantities seem to be only equal with integers. So I had to refactor the tests to check unit and magnitude separately.

khaeru reviewed (approved):
> Also, I added tests for the new metrics, which failed because Quantities seem to be only equal with integers. So I had to refactor the tests to check unit and magnitude separately.

Great, thanks for catching that.

khaeru commented:
> - [ ] [maybe] AR6-GWP20 factors are added to the globalwarmingpotentials package openclimatedata/globalwarmingpotentials#12

We'll open a follow-up issue for this.
